app/models/queue_item.rb

```diff
@@ -0,0 +1,19 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+
+  validates :video, uniqueness: true
```

Comment:
Shouldn't this be scoped to the user? Seems like user 2 won't be able to queue the video if user 1 has it queued. I would recommend you create a spec to replicate the scenario I proposed. If it fails, you know that you have a bug. When it passes, the bug is fixed. As long as you have the spec, the bug can't come back either (regression).